protocol/vm/vm_test.go

```diff
@@ -1,4 +1,4 @@
-package vm
+package vm_test
```

Comment:
I see this is mentioned in the commit message

> Some tests in protocol/vm depend on protocol/bc and have been moved into the new vm_test package to break the dependency cycle.

The ideal way to handle this would be to make the tests not depend on protocol/bc at all, and have them set up context objects directly. Do you consider that future work? Should it be mentioned in a TODO or in the commit message? Or would it be easy enough to do right here in this PR?